staging/src/k8s.io/apiserver/pkg/util/flowcontrol/match_test.go

```diff
@@ -120,162 +120,236 @@ func TestLiterals(t *testing.T) {
-			Resources:    []string{"*"},
-			ClusterScope: true}}}})
+			Verbs:         []string{"goodverb"},
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"*"},
+			ClusterScope:  true,
+			ResourceNames: []string{"goodn"},
+		}}},
+		{
+			Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
+				User: &fcv1a1.UserSubject{"goodu"}}},
+			ResourceRules: []fcv1a1.ResourcePolicyRule{{
+				Verbs:         []string{"goodverb"},
+				APIGroups:     []string{"goodapig"},
+				Resources:     []string{"goodrscs"},
+				ClusterScope:  true,
+				ResourceNames: []string{"*"},
+			}}},
+	})
 	checkRules(t, false, reqRU, []fcv1a1.PolicyRulesWithSubjects{{
 		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
 			User: &fcv1a1.UserSubject{"goodu"}}},
 		ResourceRules: []fcv1a1.ResourcePolicyRule{{
-			Verbs:        []string{"badverb"},
-			APIGroups:    []string{"goodapig"},
-			Resources:    []string{"goodrscs"},
-			ClusterScope: true}}}, {
+			Verbs:         []string{"badverb"},
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"goodrscs"},
+			ClusterScope:  true,
+			ResourceNames: []string{"goodn"},
+		}}}, {
 		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
 			User: &fcv1a1.UserSubject{"goodu"}}},
 		ResourceRules: []fcv1a1.ResourcePolicyRule{{
-			Verbs:        []string{"goodverb"},
-			APIGroups:    []string{"badapig"},
-			Resources:    []string{"goodrscs"},
-			ClusterScope: true}}}, {
+			Verbs:         []string{"goodverb"},
+			APIGroups:     []string{"badapig"},
+			Resources:     []string{"goodrscs"},
+			ClusterScope:  true,
+			ResourceNames: []string{"goodn"},
+		}}}, {
 		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
 			User: &fcv1a1.UserSubject{"goodu"}}},
 		ResourceRules: []fcv1a1.ResourcePolicyRule{{
-			Verbs:        []string{"goodverb"},
-			APIGroups:    []string{"goodapig"},
-			Resources:    []string{"badrscs"},
-			ClusterScope: true}}}, {
+			Verbs:         []string{"goodverb"},
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"badrscs"},
+			ClusterScope:  true,
+			ResourceNames: []string{"goodn"},
+		}}}, {
 		Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
 			User: &fcv1a1.UserSubject{"goodu"}}},
 		ResourceRules: []fcv1a1.ResourcePolicyRule{{
-			Verbs:        []string{"goodverb"},
-			APIGroups:    []string{"goodapig"},
-			Resources:    []string{"goodrscs"},
-			ClusterScope: false}}},
+			Verbs:         []string{"goodverb"},
+			APIGroups:     []string{"goodapig"},
+			Resources:     []string{"goodrscs"},
+			ClusterScope:  false,
+			ResourceNames: []string{"goodn"},
+		}}},
+		{
+			Subjects: []fcv1a1.Subject{{Kind: fcv1a1.SubjectKindUser,
+				User: &fcv1a1.UserSubject{"goodu"}}},
+			ResourceRules: []fcv1a1.ResourcePolicyRule{{
+				Verbs:         []string{"goodverb"},
+				APIGroups:     []string{"goodapig"},
+				Resources:     []string{"goodrscs"},
+				ClusterScope:  false,
+				ResourceNames: []string{"goodn"},
```

Comment:
(a) the indentation in this test case is different than in the others.

(b) this test case has the same values as the preceding one.

(c) we are missing a test case where there is a mismatch on the resource name.

Reply:
updated